#2/merge apps

Ianel opened the pull request:
Merge scan app and member app

Ianel committed:
feat: create checkin component

Ianel committed:
feat: add checkin route

Ianel committed:
feat: hideMainLayout on /checkin route

Ianel committed:
bug(qr-scanner): change qr scanner package

Ianel committed:
bug(checkin):  use html5-qrcode package for qr code scan

there are bugs like the size of the scan zone not being correct and zoom problem as well. might be using a more appropriated package.

Ianel committed:
feat(scan): implement scan functionality

attempt to create a qr scanner component by testing qr scanner package

Ianel committed:
feat(scan): adapt layout for checkin route

change layout to voted design

Ianel committed:
feat(stepper): implement stepper functionality

create stepper according to the design voted

Ianel committed:
design(scan): implement figma design details

Ianel committed:
bugfix: remove duplicate scrollbar

Ianel committed:
feat: implement toast functionality

attempt to use existing toast functionality but not working.

Ianel committed:
feat: better UX for checkin and connection to backend

add skeleton for activity. connect frontend to backend with real data from and to the API

Efabien commented on src/app/components/stepper/stepper.component.ts:
```diff
@@ -0,0 +1,214 @@
+  activityService = inject(ActivityService);
+  private toastService = inject(ToastService);
+
+  activitiesToDisplay = [
```
I thought  activities were data coming from the backend

Ianel committed:
fix: mise à jour des endpoints API conformément au refactor du backend

Permet au frontend d'effectuer des appels API  suite à la mise à jour du backend. Endpoints:
- activity => activities
- member => members
- checkin => checkins

Ianel committed:
fix: modification de l'envoi  du token vers l'API  suite à la mise à jour du backend

Permet au frontend d'injecter le token pour les routes protegées du backend

Ianel committed:
fix: adaptation de la logique et changement de design

Permet à l'application web de devenir plus responsive

Ianel committed:
feat: ajout de loader pour /checkin

Ajoute un loader pour la route checkin

Ianel committed:
feat(navbar): add checkin nav link item

Ianel committed:
Merge branch 'develop' into #2/merge-apps
